### PR TITLE
Task 3996 - Material UI: Refactor GridList

### PR DESCRIPTION
**Pull request by b14ckster**

### CODE REVIEW

File: jdi-light-material-ui-tests/src/test/java/io/github/epam/material/tests/layout/GridListTests.java [hunk 1]
```diff
@@ -4,47 +4,41 @@
...
-    @Test
-    public static void singleLineGridTest() {
+    @Test(dataProviderClass = GridListDataProvider.class, dataProvider = "gridListItems")
+    public static void singleLineGridTest(int number, String name) {
+        singleLineGridList.show();
         singleLineGridList.has().size(5);
-        singleLineGridList.has().image(3).and().altImgName(3, HATS);
-        singleLineGridList.has().image(4).and().altImgName(4, BIKE);
-        singleLineGridList.has().title(5, CAMERA);
+        singleLineGridList.has().image(number)
```
Comment (annav00) [inline jdi-light-material-ui-tests/src/test/java/io/github/epam/material/tests/layout/GridListTests.java:32]: Can be combined into one check via and()

File: jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/asserts/layout/GridListAssert.java [hunk 2]
```diff
@@ -7,11 +7,14 @@
...
 
+/**
+ * Assertions for {@link GridList}.
+ */
 public class GridListAssert extends UISelectAssert<GridListAssert, GridList> implements HasAssert<GridListAssert> {
 
     @JDIAction("Assert that '{name}' element {0} has title '{1}'")
     public GridListAssert title(int elNum, String reqTitle) {
-        jdiAssert(element().getTitle(elNum), Matchers.is(reqTitle));
+        jdiAssert(element().title(elNum), Matchers.is(reqTitle));
```
Comment (annav00) [inline jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/asserts/layout/GridListAssert.java:17]: Maybe it's better not to abbreviate variable names `elNum` and `reqTitle`?
The same goes for `altImgName()`.

File: jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/elements/layout/GridList.java [hunk 3]
```diff
@@ -3,32 +3,35 @@
...
+    public String altImgName(int numEl) {
+        return get(numEl).core().find("img").attr("alt");
     }
 
-    @JDIAction("Get {name} image alternative name")
-    public String getAltImgName(int numEl) {
-        return get(numEl).find("img").attr("alt");
+    @JDIAction("Get '{name}' title")
+    public String title(int numEl) {
+        return get(numEl).core().find("div[class*='title']").text().replaceAll("[\\t\\n\\r]+", " ");
```
Comment (annav00) [inline jdi-light-material-ui/src/main/java/com/epam/jdi/light/material/elements/layout/GridList.java:30]: - `numEl` and `index` have the same These words have the same meaning in the context of these methods. It is possible to use the same variable name in all cases.
- I think it's worth adding information about the image index to the action description.
-  Is it really necessary to use `core()` after `get()` here?

**Review by GrebNat — APPROVED** [review]
I am ok if all previous comments are fixed

**Pull request merged**